File: reviewly-frontend/src/app/service/ApplicationProperties.ts
import {LogLevel} from "../utils/Logger";

class ApplicationProperties {

    get basePath(): string {
        return "/app";
    }

    get apiUrl(): string {
        return process.env.API_URL;
    }

    get runMode(): string {
        return process.env.RUN_MODE;
    }

    get landingPageUrl(): string {
        return process.env.LANDING_URL;
    }

    get reviewWebLinkUrl(): string {
        return process.env.REVIEW_WEB_LINK_URL;
    }

    get authLoginPageUrl(): string {
        return process.env.AUTH_LOGIN_PAGE_URL;
    }

    get authLogoutPageUrl(): string {
        return process.env.AUTH_LOGOUT_PAGE_URL;
    }

    get authTokenUrl(): string {
        return process.env.AUTH_TOKEN_URL;
    }

    get authRedirectUri(): string {
        return process.env.AUTH_REDIRECT_URI;
    }

    get selfHost(): string {
        return process.env.SELF_HOST;
    }

    get clientId(): string {
        return process.env.CLIENT_ID;
    }

    get clientSecret(): string {
        return process.env.CLIENT_SECRET;
    }

    get downloadExtensionChrome(): string {
        return process.env.DOWNLOAD_EXTENSION_CHROME;
    }

    get chromeExtensionId(): string {
        return process.env.CHROME_EXTENSION_ID;
    }

    get userAvatarsStorageUrl(): string {
        return process.env.USER_AVATARS_URL;
    }

    get sitesIconsStorageUrl(): string {
        return process.env.SITES_ICONS_URL;
    }

    get defaultUserAvatarFileName(): string {
        return process.env.DEFAULT_USER_AVATAR_FILE_NAME;
    }

    get defaultSiteIconFileName(): string {
        return process.env.DEFAULT_SITE_ICON_FILE_NAME;
    }

    get contactEmail(): string {
        return process.env.CONTACT_EMAIL;
    }

    get logLevel(): LogLevel {
        return LogLevel[process.env.LOG_LEVEL as keyof typeof LogLevel];
    }
}

export default new ApplicationProperties();

File: reviewly-backend/src/main/java/space/reviewly/backend/config/ApplicationProperties.java
package space.reviewly.backend.config;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.net.URISyntaxException;
import lombok.Getter;
import lombok.Setter;
import org.apache.hc.core5.net.URIBuilder;
import org.springframework.boot.context.properties.ConfigurationProperties;

import java.util.List;
import org.springframework.stereotype.Component;

/**
 * Custom API application properties.
 *
 * @author Shcherbachenya Igor
 */
@Getter
@Setter
@Component("applicationProperties")
@ConfigurationProperties(prefix = "reviewly")
public class ApplicationProperties {

    public static final String API_ROOT_PATH = "/api";

    @NotNull
    private RunMode runMode;

    @NotBlank
    private String selfHost;

    @NotBlank
    private String feApplicationUrl;

    @NotBlank
    private String authApplicationUrl;

    @NotBlank
    private String authLoginPageUrl;

    @NotBlank
    private String authLogoutPageUrl;

    @NotBlank
    private String authRedirectUri;

    @NotBlank
    private String restorePasswordPageUrl;

    @NotBlank
    private String signUpPageUrl;

    @NotBlank
    private String profilePageUrl;

    @NotBlank
    private String clientId;

    @NotBlank
    private String clientSecret;

    @NotBlank
    private String downloadExtensionChrome;

    @NotBlank
    private String chromeExtensionId;

    @NotBlank
    private String reviewlyInfoEmail;

    @NotBlank
    private String noReplyEmail;

    @NotBlank
    private String contactEmail;

    @NotBlank
    private String fileStorageUrl;

    @NotBlank
    private String fileStorageUrlSitesIcons;

    @NotBlank
    private String fileStorageUrlUsersAvatars;

    @NotBlank
    private String defaultUserAvatarFileName;

    @NotBlank
    private String defaultSiteIconFileName;

    private List<String> allowedOrigins;

    public String getLoginUrl() throws URISyntaxException {
        URIBuilder loginUrlBuilder = new URIBuilder(this.authLoginPageUrl);
        loginUrlBuilder.addParameter("response_type", "code");
        loginUrlBuilder.addParameter("client_id", clientId);
        loginUrlBuilder.addParameter("redirect_uri", authRedirectUri);

        return loginUrlBuilder.build().toString();
    }
}